contracts/extensions/lending/ApeSwapZapLending.sol

```diff
@@ -0,0 +1,55 @@
+// SPDX-License-Identifier: GPL-3.0-only
+pragma solidity ^0.8.0;
+
+import "./lib/ICErc20.sol";
+import "../../utils/TransferHelper.sol";
+
+abstract contract ApeSwapZapLending is TransferHelper {
+    using SafeERC20 for IERC20;
+    using SafeERC20 for ICErc20;
+
+    /// @dev Native token market underlying
+    address public constant LENDING_NATIVE_UNDERLYING = 0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE;
+
+    event ZapLending(uint256 inputAmount, ICErc20 market, uint256 outputAmount);
+    event ZapLendingMarket(uint256 inputAmount, ICErc20 market);
+
+    /// @notice Zap token single asset lending market
+    /// @param inputAmount Amount of input tokens to zap
+    /// @param market Lending market to deposit to
+    /// @param recipient Recipient of cTokens
+    function zapLending(uint256 inputAmount, ICErc20 market, address recipient) external payable {
+        IERC20 underlyingToken = IERC20(market.underlying());
+        inputAmount = _transferIn(underlyingToken, inputAmount);
+
+        if (address(underlyingToken) == LENDING_NATIVE_UNDERLYING) {
+            uint256 depositAmount = msg.value;
+            market.mint{value: depositAmount}();
```

Comment:
I think this section needs some more thought.

1. `inputAmount = _transferIn(underlyingToken, inputAmount);` The underlying token might be `0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE` which would revert the transfer in right?
2. `uint256 depositAmount = msg.value;` What if we are multicalling `zapLending` after a swap and want to deposit `WNative`

Reply:
1. Moved `transferIn()` into else{} when not `NATIVE_UNDERLYING`
2. `uint256 depositAmount = inputAmount == 0 ? address(this).balance : inputAmount;` ? Now it can swap, unwrap and zap to lending with either custom amount or everything. (Yes this does allow for 'stealing' any native that in there. which there shouldn't, so not 100% sure of best approach)

Reply:
Nice, the approach looks good to me.

For this case I would think that using the constant would be ideal, otherwise it follows a similar pattern as `_tranferIn`
`inputAmount == Constants.CONTRACT_BALANCE ? address(this).balance : inputAmount`